cpp/tests/logger_tests.cpp

```diff
@@ -105,20 +105,19 @@ void expect_log_events(std::string const& filename,
 {
   auto actual_events = rmm::detail::parse_csv(filename);
 
-  std::equal(expected_events.begin(),
-             expected_events.end(),
-             actual_events.begin(),
-             [](auto expected, auto actual) {
-               // We don't test the logged thread id since it may be different from what we record.
-               // The actual value doesn't matter so long as events from different threads have
-               // different ids
-               // EXPECT_EQ(expected.thread_id, actual.thread_id);
-               // EXPECT_EQ(expected.stream, actual.stream);
-               EXPECT_EQ(expected.act, actual.act);
-               EXPECT_EQ(expected.size, actual.size);
-               EXPECT_EQ(expected.pointer, actual.pointer);
-               return true;
-             });
+  EXPECT_TRUE(std::equal(expected_events.begin(),
```

Comment:
This test design doesn't make a lot of sense. We don't need the equality-checking behavior of `std::equal` because the item-for-item equality is already checked in the lambda. The `std::equal` here is basically being used as a control structure like a for loop rather than an algorithm with a result...

Reply:
I agree that it doesn't make much sense, I just didn't want to go through the trouble of adding in a zip iterator in this PR. I can do that if you feel strongly though.

Reply:
We should refactor this. A separate PR would be best since it's not especially related to GCC 14.

Reply:
https://github.com/rapidsai/rmm/pull/1999